src/go/rpk/pkg/cli/cluster/partitions/list.go

```diff
@@ -1,207 +1,231 @@
-					tw := out.NewTable(headers...)
-					for _, rs := range p.ReconciliationStatuses {
-						var row []interface{}
-						row = append(row, rs.NodeID)
-						for _, s := range rs.Operations {
-							row = append(row, s.Core, s.Type, s.RetryNumber, s.Revision, s.Status)
-						}
-						tw.Print(row...)
+			var clusterPartitions []adminapi.ClusterPartition
+			var mu sync.Mutex
+			if len(topics) == 0 && all {
+				clusterPartitions, err = cl.AllClusterPartitions(cmd.Context(), true, disabledOnly)
+				// If the admin API returns a 404, most likely rpk is talking
+				// with an old cluster.
+				if he := (*adminapi.HTTPResponseError)(nil); errors.As(err, &he) {
+					if he.Response.StatusCode == http.StatusNotFound {
+						out.Die("unable to query all partitions in the cluster: %v You may need to upgrade the cluster to access this feature or try listing per-topic with: 'rpk cluster partition list [TOPICS]'", err)
 					}
-					tw.Flush()
-					j++
-					if j < len(response) {
-						fmt.Println()
+				}
+				out.MaybeDie(err, "unable to query all partitions in the cluster: %v", err)
+			} else {
+				g, egCtx := errgroup.WithContext(cmd.Context())
+				for _, topic := range topics {
+					if topic == "" {
+						out.Die("invalid empty topic\n%v", cmd.UsageString())
 					}
+					ns, topicName := nsTopic(topic)
+					g.Go(func() error {
+						cPartition, err := cl.TopicClusterPartitions(egCtx, ns, topicName, disabledOnly)
+						if err != nil {
+							// If the admin API returns a 404, most likely rpk
+							// is talking with an old cluster.
+							var he *adminapi.HTTPResponseError
+							isNotFoundErr := errors.As(err, &he) && he.Response.StatusCode == http.StatusNotFound
+							if !isNotFoundErr {
+								return fmt.Errorf("unable to query cluster partition metadata of topic %q: %v", topicName, err)
+							}
+							zap.L().Sugar().Warn("Admin API 'GET /v1/cluster/partitions' returned 404, trying now with 'GET /v1/partitions'")
+							if disabledOnly {
+								zap.L().Sugar().Warn("--only-disabled flag is not supported in this API version")
+							}
+							cPartition, err = topicPartitions(egCtx, cl, ns, topicName)
+							if err != nil {
+								return err
+							}
+						}
+						mu.Lock()
+						clusterPartitions = append(clusterPartitions, cPartition...)
+						mu.Unlock()
+						return nil
+					})
 				}
-			})
+				err := g.Wait()
+				out.MaybeDieErr(err)
+			}
+			if partitions != nil {
+				clusterPartitions = filterPartition(clusterPartitions, partitions)
+			}
+			printClusterPartitions(f, clusterPartitions)
 		},
 	}
+	cmd.Flags().BoolVarP(&all, "all", "a", false, "If true, list all partitions in the cluster")
+	cmd.Flags().BoolVar(&disabledOnly, "only-disabled", false, "If true, list disabled partitions only")
+	cmd.Flags().IntSliceVarP(&partitions, "partition", "p", nil, "Partitions to show current assignments (repeatable)")
```

Comment:
```suggestion
	cmd.Flags().IntSliceVarP(&partitions, "partition", "p", nil, "Partitions to show current assignments (repeatable)")
```

This could stand to be clearer.

Reply:
Changed to `List of comma-separated partitions IDs that you wish to filter the results with`